### ISSUE TITLE
`expect_match()` documentation

### ISSUE THREAD

**Issue by maia-sh** [r-lib/testthat#1319]
Thanks for another great package! I'm new to {testthat} and am having trouble sorting through the documentation on `expect_match()`. I'd like to make my tests in line with the [3rd edition](https://testthat.r-lib.org/articles/third-edition.html), and I'm unclear as to whether `expect_match()` is being soft-deprecated. It's not listed in the [Reference](https://testthat.r-lib.org/reference/index.html) but I'm not seeing any deprecation flag on the [function documentation](https://testthat.r-lib.org/reference/expect_match.html). I also see it in the latest [R packages](https://r-pkgs.org/tests.html#expectations) but then see the function was [marked as internal](https://github.com/r-lib/testthat/commit/6325a41426b5495e68e8d3773dce095e73bfafd9).

In short, should new tests avoid using `expect_match()` and instead use `expect_identical()`, `expect_equal()`, or `expect_true()`?

Thank you!

**Comment by hadley** [comment]
Sorry, why do you think it's being deprecated?

**Comment by maia-sh** [comment]
I was confused about the documentation, since the function is now marked `@keywords internal` and is therefore no longer in the package index. However, I'm now seeing from [your book](https://r-pkgs.org/man.html#roxygen-comments) that this is a flag for "functions that are of interest to other developers extending your package, but not most users."
So my understanding now is that `expect_match()` is not being deprecated and is fine to use in 3rd edition tests but could also be substituted by other functions for most users not extending the package.
I think that solves the issue, so I'll close it. Thank you!

**Comment by hadley** [comment]
In general now, I think you're better off using `expect_snapshot()` or similar. `expect_match()` is confusing because it uses regular expressions (it's not just a raw text match), and it's in general hard to construct a good regexp that won't fail at some point down the road.
